### DISCUSSION TITLE
producing a homodimer

### DISCUSSION THREAD

**Discussion by victoriapopov** [chaidiscovery/chai-lab#242] [Q&A] [answered]
About 2 weeks ago i managed to model homodimers by providing the same sequence twice via python. Now it is not working and I'm getting the error- "raise ValueError("Duplicate entity %s found" % entity)".
Why has it changed and how can I model homodimers now?

**Comment by arogozhnikov** [comment, marked as answer]
You just need to give them different names, like so:

```
example_fasta = """
>protein|name=peptide-1
AGSHSMRYFST
>protein|name=peptide-2
AGSHSMRYFSTDM
"""

**Reply by victoriapopov** [reply]
I did give them different names, I'm also using the same script that worked for me 2 weeks ago and suddenly it comes up with this error. I'm using a csv file to create a fasta.
`>protein|{var_name+'a'}
{seq1}
>protein|{var_name+'b'}
{seq2}`

**Reply by arogozhnikov** [reply]
I don't see where/why it would come from then. 

I'm not sure this error comes from our code then - we don't have `Duplicate entity` is the codebase, so I need your stacktrace + output of `pip freeze` to make any conclusion. Better yet, an example of failing input.

Side comment: new input format uses `name=`, this was changed more than a month ago (sep 20), not two weeks

**Reply by victoriapopov** [reply]
Thanks for the feedback,
I'm attaching both the output from `pip freeze` and the error log.
[requirements.txt](https://github.com/user-attachments/files/18111305/requirements.txt)
[err.txt](https://github.com/user-attachments/files/18111537/err.txt)

**Reply by arogozhnikov** [reply]
thanks, we just fixed it, please try installing fresh version with 
```
pip install git+https://github.com/chaidiscovery/chai-lab.git
```

**Reply by victoriapopov** [reply]
Hi,
thanks a lot for the reply.
unfortunately even after updating the version I still encounter the same issue.
I also tried running the example fasta from #245 
and still got "ValueError: Duplicate entity <ihm.Entity(peptide-2)> found".
[error.txt](https://github.com/user-attachments/files/18140268/error.txt)

**Reply by arogozhnikov** [reply]
example from #245 works on main (there is no `peptide-2` among names). Example from #251 also works on main. Pls check you have fresh version from the main branch.
